app/models/course/condition/assessment.rb

```diff
@@ -0,0 +1,45 @@
+class Course::Condition::Assessment < ActiveRecord::Base
+  include ActiveSupport::NumberHelper
+
+  acts_as_condition
+  belongs_to :assessment, class_name: Course::Assessment.name, inverse_of: false
+
+  default_scope { includes(:assessment) }
+
+  def title
+    if minimum_grade_percentage
+      minimum_grade_percentage_display = number_to_percentage(minimum_grade_percentage,
+                                                              precision: 2,
+                                                              strip_insignificant_zeros: true)
+      self.class.human_attribute_name('title.title',
+                                      assessment_title: assessment.title,
+                                      minimum_grade_percentage: minimum_grade_percentage_display)
+    else
+      assessment.title
+    end
+  end
+
+  def satisfied_by?(course_user)
+    assessment = assessment_with_submissions_by_user(assessment_id, course_user.user)
+    return false unless assessment
+
+    graded_submission = assessment.submissions.find(&:graded?)
+    return false unless graded_submission
+
+    if minimum_grade_percentage
+      100.0 * graded_submission.grade / assessment.maximum_grade >= minimum_grade_percentage
+    else
+      true
+    end
+  end
+
+  private
+
+  def assessment_with_submissions_by_user(assessment_id, user)
+    Course::Assessment.
```

Comment:
Separate PR, get that merged, then fix this. The condition is complex and not obvious.